src/psfmachine/ffi.py

```diff
@@ -0,0 +1,924 @@
+from astropy.time import Time
+from astropy.wcs import WCS
+from photutils import Background2D, MedianBackground, BkgZoomInterpolator
+
+# from . import PACKAGEDIR
+from .utils import do_tiled_query
+from .machine import Machine
+from .version import __version__
+
+__all__ = ["FFIMachine"]
+
+
+class FFIMachine(Machine):
+    """
+    Subclass of Machine for working with FFI data. It is a subclass of Machine
+    """
+
+    def __init__(
+        self,
+        channel=1,
+        quarter=5,
+        wcs=None,
+        limit_radius=32.0,
+        n_r_knots=10,
+        n_phi_knots=15,
+        cut_r=6,
+        rmin=1,
+        rmax=16,
+        **kwargs,
+    ):
+        """
+        Class to work with FFI data.
+
+        Parameters
+        ----------
+        channel : int
+            Channel number to be used
+        quarter : int
+            Quarter/Campagn nunmber to be used (for Kepler data).
+        wcs : astropy.wcs
+            World coordinates system solution for the FFI. Used for plotting.
+        **kwargs
+            Keyword attributes that contain information parsed from `from_file()` and
+            is used to initialize a `Machine` class object.
+
+        Attributes
+        ----------
+        All attributes inherited from Machine.
+
+        meta : dictionary
+            Meta data information related to the FFI
+        wcs : astropy.wcs
+            World coordinates system solution for the FFI. Used for plotting.
+        flux_2d : numpy.ndarray
+            2D image representation of the FFI, used for plotting.
+        channel : int
+            Channel number to be used
+        quarter : int
+            Quarter/Campagn nunmber to be used (for Kepler data).
+        """
+        self.column = kwargs["column"].ravel()
+        self.row = kwargs["row"].ravel()
+        self.ra = kwargs["ra"].ravel()
+        self.dec = kwargs["dec"].ravel()
+        # keep 2d image for easy plotting
+        self.flux_2d = kwargs["flux"]
+        # reshape flux and flux_err as [ntimes, npix]
+        self.flux = kwargs["flux"].reshape(kwargs["flux"].shape[0], -1)
+        self.flux_err = kwargs["flux_err"].reshape(kwargs["flux_err"].shape[0], -1)
+        self.sources = kwargs["sources"]
+
+        # remove background and mask bright/saturated pixels
+        # these steps need to be done before `machine` init, so sparse delta
+        # and flux arrays have the same shape
+        self._remove_background()
+        self._mask_pixels()
+
+        # init `machine` object
+        super().__init__(
+            kwargs["time"],
+            self.flux,
+            self.flux_err,
+            self.ra,
+            self.dec,
+            self.sources,
+            self.column,
+            self.row,
+            n_r_knots=n_r_knots,
+            n_phi_knots=n_phi_knots,
+            cut_r=cut_r,
+            rmin=rmin,
+            rmax=rmax,
+        )
+        self.meta = kwargs["metadata"]
+        self.channel = channel
```

Comment:
This is where we could be more flexible and have some if statements for if we've passed any of 
* quarter
* campaign
* sector
* channel
* camera
* ccd